### PR TITLE
Code optimizations reported by static code analysis (2019-01-11)

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

SUMMARY: Performance "Code optimizations reported by static code analysis"

#### Purpose of change

Cleanup code while increase performance and fix minor error.

#### Describe the solution

See commit messages.

### CODE REVIEW

File: src/skill.h [hunk 1]
```diff
@@ -79,7 +79,7 @@ class SkillLevel
         int _highestLevel = 0;
 
     public:
-        SkillLevel() {}
+        SkillLevel() = default;
```
Comment (ZhilkinSerg) [inline src/skill.h]:
```suggestion
        SkillLevel() {}
```

**Comment by ZhilkinSerg** [comment]
> 
> 
> Travis `CLANG=clang++-3.8` has failed with
> 
> ```c++
> src/skill.cpp:225:29: error: default initialization of an object of const type
>       'const SkillLevel' without a user-provided default constructor
>     static const SkillLevel null_skill;
>                             ^
>                                       {}
> ```

Thanks, fixed that.

**Comment by ZhilkinSerg** [comment]
Jenkins is failing due to changes in `json.cpp` and `json.h`.

File: src/explosion.cpp [hunk 2]
```diff
@@ -321,7 +321,9 @@ void game::explosion( const tripoint &p, const explosion_data &ex )
             int qty = shr.casing_mass * std::min( 1.0, shr.recovery / 100.0 ) /
                       to_gram( fragment_drop->weight );
             // Truncate to a random selection
-            std::random_shuffle( tiles.begin(), tiles.end() );
+            static auto eng = std::default_random_engine(
+                                  std::chrono::system_clock::now().time_since_epoch().count() );
+            std::shuffle( tiles.begin(), tiles.end(), eng );
```
Comment (jbytheway) [inline src/explosion.cpp]: This change is going to make explosions less reproducible, which might lead to some annoying-to-pin-down test failures.  But there's already other `shuffle` calls with the same issue; probably makes sense to fix them all at once.

Comment (ZhilkinSerg) [reply src/explosion.cpp]: I've just changed `std::random_shuffle` to `std:shuffle` everywhere. That was the last unchanged instance.

Do you think we need to change random engine seed from system clock to something else ?

**Pull request merged**